text/0005-media-webrtc-sdk.md

```diff
@@ -0,0 +1,589 @@
+After several attempts (configurable), the client will stop retrying and report an error to the user as CONNECTION_TIMEOUT.
+
+## 4.3 Restart-ice
+
+**_Endpoint_**: POST `GATEWAY/webrtc/:conn_id/restart-ice`
+
+**_Headers_**: `Authorization: Bear {token}`
+
+**_Body_**: same with connect request but the tracks should sending with current state
+
+**_Response Data_**: same with connect response
+
+
+By doing this, in case of a network change, the client can retry connecting to the server with the newest offer SDP. If the server is still alive, it will respond with a new answer SDP. However, if the server is dead, the gateway will retry connecting to another server. The session state can be restored using the track state and each feature state.
+
+## 4.4 Ice-tricle
+
+Each time the client's WebRTC connection has a new ice-candidate, it should be sent to the gateway using the following endpoint:
+
+**_Endpoint_**: POST `GATEWAY/webrtc/conns/:conn_id/ice-remote`
+
+**_Body_**: candidate String
+
+**_Response Data_**: None
+
+## 4.5 Datachannel Request/Response format
+
+The format for encoding all requests and responses sent over the data channel is JSON. The structure of the request and response objects is as follows:
+
+Request/Event:
+
+```
+{
+    type: "event" | "request",
+    seq: Number,
+    cmd: String,
+    data: Option<JSON>,
+}
+```
+
+Response:
+
+```
+{
+    type: "answer",
+    seq: Number,
+    success: bool,
+    error_code: Option<String>,
+    error_msg: Option<String>,
+    data: Option<JSON>,
+}
+```
+
+The seq is an incremental value generated on the sender side. It helps us to map between requests and responses and also detect data loss.
+
+The cmd is generate with rule: `identify.action`, for example:
+
+- `peer.updateSdp`.
+- `sender.{sender_id}.toggle`.
+- `receiver.{receiver_id}.switch`.
+- `room.peers.subscribe`.
+- `room.peers.unsubscribe`.
+- `session.disconnect`.
+- `session.features.mix-minus.add_source`.
+
+## 4.6 In-session requests
+
+
+At the current state, we only have one WebRTC connection to the server, so there is no need to send any requests over HTTP. All requests will be sent over the WebRTC datachannel.
+
+Typically, the client will need to perform various actions with the media server, such as:
+
+- Creating/Releasing senders
+- Creating/Releasing receivers
+- Sender actions: pause, resume, switch stream
+- Receiver actions: pause, resume, switch remote source, update priority, and layers
+
+All actions that involve changing tracks will be performed locally first, and then the `updateSdp` command will be sent to the server.
+
+### 4.6.1 UpdateSDP
+
+Each time we make changes to the WebRTC connection or negotiationneeded event fired, we need to send an `updateSdp` request to the server over the data channel. This request is described below:
+
+**_Cmd:_**: `peer.updateSdp`
```

Comment:
inconsistence in naming cases, "mix-minus" , "add_source", "updateSdp" are in different cases. The "mix-minus" is fine I guess, but this should be in snake case